Lib/test/test_xxlimited.py

```diff
@@ -1,19 +1,39 @@
 import unittest
 from test.support import import_helper
-import types
 
 xxlimited = import_helper.import_module('xxlimited')
-xxlimited_35 = import_helper.import_module('xxlimited_35')
 
-
-class CommonTests:
-    module: types.ModuleType
-
-    def test_xxo_new(self):
-        xxo = self.module.Xxo()
-
-    def test_xxo_attributes(self):
-        xxo = self.module.Xxo()
+# if import of xxlimited succeeded, the other ones should be importable.
+import xxlimited_3_13
+import xxlimited_35
+
+MODULES = {
+    (3, 15): xxlimited,
+    (3, 13): xxlimited_3_13,
+    (3, 5): xxlimited_35,
+}
+
+def test_with_xxlimited_modules(since=None, until=None):
+    def _decorator(func):
+        def _wrapper(self, *args, **kwargs):
+            for version, module in MODULES.items():
+                if since and version < since:
+                    continue
+                if until and version >= until:
+                    continue
+                with self.subTest(version=version):
```

Comment:
I assume we want `module` in the subtest as well:
```suggestion
                with self.subTest(version=version, module=module):
```

Reply:
This is printed out as context when the subtest fails. We don't need the extra `<module ...>`; the version is enough.